### PR TITLE
[Fixes #1428] Improve Sonar Cloud security hotspots metrics 

### LINKED ISSUE #1428: Improve Sonar Cloud security hotspots metrics
**Describe the bug**
Fix security hotspots
https://sonarcloud.io/project/security_hotspots?id=provectus_kafka-ui_frontend&sinceLeakPeriod=true

**Expected behavior**
Security Hotspots Metric should be 100%

### PR DESCRIPTION
**Pull request by Motherships**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing applications:)

**What changes did you make?** (Give an overview)
- Fixes #1428 
- Replaced risky regexes by similar constructions

**Is there anything you'd like reviewers to focus on?**
- May be I need to write tests to test this?
- I don't really like how `kafka-ui-react-app/src/redux/reducers/loader/reducer.ts` looks

**How Has This Been Tested?** (put an "X" next to an item)

- [X] No need to
- [ ] Manually(please, describe, when necessary)
- [ ] Unit checks
- [ ] Integration checks
- [X] Covered by existing automation

**Checklist** (put an "X" next to an item, otherwise PR will fail)
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [X] My changes generate no new warnings(e.g. Sonar is happy)
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

### CODE REVIEW

File: kafka-ui-react-app/src/components/Version/__tests__/compareVersions.spec.ts [hunk 1]
```diff
@@ -63,10 +60,15 @@ describe('compareVersions function', () => {
   });
 
   it('returns valid result (negative test cases)', () => {
+    // @ts-expect-error first arg is number
```
[the remark this answers is not in the document]
Comment (Motherships) [reply kafka-ui-react-app/src/components/Version/__tests__/compareVersions.spec.ts:63]: It was there before

File: kafka-ui-react-app/src/components/Version/compareVersions.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 const split = (v: string): string[] => {
-  const c = v.replace(/^v/, '').replace(/\+.*$/, '');
-  return c.split('-')[0].split('.');
+  const c = v.replace('v', '').split('-')[0];
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Version/compareVersions.ts:2]: 👍

File: kafka-ui-react-app/src/redux/reducers/alerts/reducer.ts [hunk 3]
```diff
@@ -9,8 +9,7 @@ export const initialState: AlertsState = {};
 const reducer = (state = initialState, action: Action): AlertsState => {
   const { type } = action;
 
-  const matches = /(.*)__(FAILURE)$/.exec(type);
-  if (matches && matches[2]) return addError(state, action);
+  if (type.endsWith('__FAILURE')) return addError(state, action);
```
Comment (workshur) [inline kafka-ui-react-app/src/redux/reducers/alerts/reducer.ts:12]: 👍

**Review by workshur — APPROVED** [review]
LGTM

File: kafka-ui-react-app/src/components/Version/__tests__/compareVersions.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Motherships) [reply kafka-ui-react-app/src/components/Version/__tests__/compareVersions.spec.ts:63]: W/e rewrote this part

**Review by NelyDavtyan — APPROVED** [review]

**Review by Hurenka — APPROVED** [review]

**Pull request merged**